frame/staking/src/pallet/impls.rs

```diff
@@ -1152,3 +1211,70 @@ where
 		consumed_weight
 	}
 }
+
+impl<T: Config> VoteWeightProvider<T::AccountId> for Pallet<T> {
+	fn vote_weight(who: &T::AccountId) -> VoteWeight {
+		Self::weight_of(who)
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn set_vote_weight_of(who: &T::AccountId, weight: VoteWeight) {
+		// this will clearly results in an inconsistent state, but it should not matter for a
+		// benchmark.
+		use sp_std::convert::TryInto;
+		let active: BalanceOf<T> = weight.try_into().map_err(|_| ()).unwrap();
+		let mut ledger = Self::ledger(who).unwrap_or_default();
+		ledger.active = active;
+		<Ledger<T>>::insert(who, ledger);
+		<Bonded<T>>::insert(who, who);
+
+		// also, we play a trick to make sure that a issuance based-`CurrencyToVote` behaves well:
+		// This will make sure that total issuance is zero, thus the currency to vote will be a 1-1
+		// conversion.
+		let imbalance = T::Currency::burn(T::Currency::total_issuance());
+		// kinda ugly, but gets the job done. The fact that this works here is a HUGE exception.
+		// Don't try this pattern in other places.
+		sp_std::mem::forget(imbalance);
+	}
+}
+
+/// A simple voter list implementation that does not require any additional pallets.
```

Comment:
more should be said about how inaccurate the best effort sorting is.

We should warn people that this implementation is probably not what they want.

Reply:
dba68757f23b59ef46fdbbb413dfcd3979a1d8ff